monai/apps/auto3d/data_analyzer.py

```diff
@@ -0,0 +1,541 @@
+        return case_stats
+
+    def _get_label_stats_summary(self):
+        """Get unique_label"""
+        case_stats_summary = {
+            "label_stats": {
+                "labels": self.label_union,
+                "pixel_percentage": self._pixelpercent_summary,
+                "image_intensity": partial(self._intensity_summary, average=self.SUMMERY_AVERAGE),
+            }
+        }
+        key_chain = ["label_stats", "labels"]
+        opt = self.label_union
+        unique_label = opt(
+            list(filter(None, [recursive_getvalue(case, key_chain) for case in self.results["stats_by_cases"]]))
+        )
+        for index in unique_label:
+            label_dict_summary = {"image_intensity": partial(self._intensity_summary, average=self.SUMMERY_AVERAGE)}
+            if self.DO_CONNECTED_COMP:
+                label_dict_summary["shape"] = partial(self._stats_opt_summary, is_label=True)
+                label_dict_summary["ncomponents"] = partial(self._stats_opt_summary, is_label=True)
+            case_stats_summary["label_stats"].update({f"label_{index}": label_dict_summary})
+        self.gather_summary.update(case_stats_summary)
+
+    def _pixelpercent_summary(self, x):
+        """
+        Define the summary function for the pixel percentage over the whole dataset
+        x: list of dictionaries dict = {'label1': percent, 'label2': percent}. The dict may miss some labels
+
+        """
+        percent_summary = {}
+        for _ in x:
+            for key, value in _.items():
+                percent_summary[key] = percent_summary.get(key, 0) + value
+        total_percent = np.sum(list(percent_summary.values()))
+        for key in percent_summary.keys():
+            percent_summary[key] = float(percent_summary[key] / total_percent)
+        return percent_summary
+
+    def _intensity_summary(self, x, average=False):
+        """
+        Define the summary function for a stats over the whole dataset
+        x: list of the list of intensity stats [[{max:, min:, },{max:, min:, }]]
+        average: if True, average the values over all the images (modalities)
+
+        """
+        result = {}
+        for key in x[0][0].keys():
+            value = []
+            for case in x:
+                value.append([_[key] for _ in case])
+            axis = (0,) if not average else None
+            try:
+                value = self.operations_summary[key](value, axis=axis)
+            except SyntaxWarning:
+                logger.debug("operation summary definition must accept axis input like np.mean")
+                warnings.warn("operation summary definition must accept axis input like np.mean")
+                pass
+            result[key] = np.array(value).tolist()
+        return result
+
+    def _stats_opt_summary(self, data, average=False, is_label=False):
+        """
+        Wraps _stats_opt for a list of data from all cases. Does not guarantee correct output
```

Comment:
I don't see this function using `_stats_opt` and just as its wrapper.

Thanks.

Reply:
It seems the _stats_opt isn't really used. Does this comment come from an older version of the code @dongyang0122 ? I rewrote the docstring like: Combine other stats calculation methods (like shape/min/max/std). Does not guarantee correct output for custimized stats structure. Check the following input structures.